### PR TITLE
fix #41 jwt strategy

### LINKED ISSUE #41: JWT strategy has an if-condition which is always true
**Describe the bug**
We have an if condition that always returns `true` [here](https://github.com/Human-Connection/Nitro-Backend/blob/master/src/jwt/strategy.js#L23). I guess we have to call neo4j on that line.

**To Reproduce**
I don't know how to reproduce a faulty behaviour. I checked if there is a different behaviour if I send
```json
{
  "Authorization": "Bearer whatever"
}
```
But there is no difference in behaviour. I guess it's because we're not checking the user at all.

**Expected behavior**
I never want to see `if(true)` anywhere in the code.

**Additional context**
Eslint gave me the error on commit f00dbb67dfec192dd470483230fb862b62cbbe9a:
```
yarn run eslint
```

### PR DESCRIPTION
**Pull request by DakshMiglani**
fix #41

### CODE REVIEW

**Comment by DakshMiglani** [comment]
Fixes #41

**Comment by DakshMiglani** [comment]
@roschaefer we can cache it using redis maybe?

File: src/jwt/strategy.js [hunk 1]
```diff
@@ -17,16 +18,26 @@ export default () => {
...
-      // TODO: fix https://github.com/Human-Connection/Nitro-Backend/issues/41
-      /* eslint-disable */
-      if (true) {
-      /* eslint-enable */
-        next(null, {})
+    async (JWTPayload, next) => {
+      const session = driver.session()
+      const result = await session.run(
+        'MATCH (user:User {id: $userId}) ' +
+        'RETURN user {.id, .slug, .name, .avatar, .email, .password, .role} as user LIMIT 1',
```
Comment (appinteractive) [inline src/jwt/strategy.js]: Please remove password here

File: src/jwt/strategy.js [hunk 2]
```diff
@@ -17,16 +18,26 @@ export default () => {
...
+          userId: JWTPayload.id
+        }
+      )
+      session.close()
+      const [currentUser] = await result.records.map((record) => {
+        return record.get('user')
+      })
+
+      if (currentUser) {
+        delete currentUser.password
```
Comment (appinteractive) [inline src/jwt/strategy.js]: you don`t need to delete the password here when you do not return it in the above query

Comment (DakshMiglani) [reply src/jwt/strategy.js]: sorry, new to neo4j. I'll edit that right now!

**Comment by DakshMiglani** [comment]
@appinteractive done!

**Comment by appinteractive** [comment]
Thanks @DakshMiglani for your valuable contribution! 🥳

**Comment by roschaefer** [comment]
@appinteractive but what problem are we fixing here? Is it that we change the user data, hit refresh and the data is not updated? Because it was taken from the JWT token?

**Comment by roschaefer** [comment]
I am struggling with this PR, I belive the strategy never gets' called. @DakshMiglani how did you debug the code?

**Comment by DakshMiglani** [comment]
@appinteractive @roschaefer the problem we are fixing here is about knowing which user made the request, passport.js doesn't automatically know which user has made request from the jwt. So here, we are using the jwt to find which user made the request and passing the user object back to passport, so that passport can make it available to our resolvers.

we can know which user has made this request, by checking `req.user` in our resolvers.

**Comment by appinteractive** [comment]
Seems that the strategy is not used in the current setup?

**Pull request merged**